hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/multimap/AbstractMultiMapAddEntryListenerMessageTask.java

```diff
@@ -102,12 +101,12 @@ public void onMapEvent(MapEvent event) {
                 final EntryEventType type = event.getEventType();
                 final String uuid = event.getMember().getUuid();
                 sendClientMessage(null, encodeEvent(null,
-                        null, type.getType(),
+                        null, null, type.getType(),
                         uuid, event.getNumberOfEntriesAffected()));
             }
         }
     }
 
-    protected abstract ClientMessage encodeEvent(Data key, Data value,
+    protected abstract ClientMessage encodeEvent(Data key, Data value, Data oldValue,
```

Comment:
It makes more sense to do the old value and then the value. E.g. if you look at the AtomicLong.compareAndSwap(old, new)

Should be very easy to change with Intellij signature change refactoring.